### PR TITLE
prometheus-node-exporter-lua: new uci_dhcp_host module

### PR DESCRIPTION
**Pull request by Tycale**
Extract data from configuration file /etc/config/dhcp and create labels
{name, ip, mac, dns} via uci. Those labels are useful in order to craft
complex prometheus queries as replacing the MAC address to custom name.
E.g.: wifi_station_signal_dbm * on (mac) group_left(name) uci_dhcp_host
or on (mac) label_replace(wifi_station_signal_dbm, "name", "$1", "mac",
"(.+)")

Signed-off-by: Gérondal Thibault <contact@tycale.be>

Maintainer:  @champtar
Compile tested: Marvell Armada 37x/38x/XP, Linksys WRT1900ACS (Shelby), v19.07.2
Run tested: Hit http://device:9100/metrics and it shows correctly the expected entries
```
# TYPE uci_dhcp_host gauge
uci_dhcp_host{mac="XX:XX:XX:XX:XX:XX",dns="1",name="XXXXX",ip="192.168.2.1XX"} 1
...
```

Description: Expose static leases informations to Prometheus. This module enables crafting complex queries where MAC addresses or IPs can be translated to a chosen name.

### CODE REVIEW

**Comment by Tycale** [comment]
> Add static forward and reverse DNS entries for this host.
(source: https://openwrt.org/docs/guide-user/base-system/dhcp#static_leases)

DNS option is for being able to query an host via the DNS server (dnsmasq) running on the OpenWRT router. 

Also, if an option does not exist (mac, name, ip or dns), the label is removed because of the behavior of Lua's table structure.

**Comment by aparcar** [comment]
@champtar I'm fine merging this, any concerns?

File: utils/prometheus-node-exporter-lua/Makefile [hunk 1]
```diff
@@ -166,6 +177,7 @@ endef
 
 $(eval $(call BuildPackage,prometheus-node-exporter-lua))
 $(eval $(call BuildPackage,prometheus-node-exporter-lua-nat_traffic))
+$(eval $(call BuildPackage,prometheus-node-exporter-lua-uci_dhcp_host))
```
Comment (aparcar) [inline utils/prometheus-node-exporter-lua/Makefile:180]: Please apply alphabetical ordering.

File: utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/uci_dhcp_host.lua [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  local curs=uci.cursor()
+  local metric_uci_host = metric("uci_dhcp_host", "gauge")
+
+  curs:foreach("dhcp", "host", function(s)
+    if s[".type"] == "host" then
+      labels = {name=s["name"], mac=string.upper(s["mac"]), dns=s["dns"], ip=s["ip"]}
+      metric_uci_host(labels, 1)
+    end
+  end)
+
```
Comment (aparcar) [inline utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/uci_dhcp_host.lua:13]: Please remove the empty line.

**Comment by champtar** [comment]
> @champtar I'm fine merging this, any concerns?

There is only 24h in a day :P, please go ahead, I'll not have any time before next week (and even then not sure)

File: utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/uci_dhcp_host.lua (hunk 2 above)
Comment (Tycale) [reply utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/uci_dhcp_host.lua]: Done

File: utils/prometheus-node-exporter-lua/Makefile (hunk 1 above)
Comment (Tycale) [reply utils/prometheus-node-exporter-lua/Makefile]: Done, plus did a separate commit to order everything 
(on top of my previous commit) https://github.com/openwrt/packages/pull/12712/commits/c91554935f5adfc46fb047aba2401231ae8a077a ; fine with this ?

**Comment by aparcar** [comment]
@Tycale Would you mind ordering the package definitions in the Makefile as well?

**Comment by Tycale** [comment]
> @Tycale Would you mind ordering the package definitions in the Makefile as well?

Ok, but I am not 100% sure how you want the file to be ordered, tell me if you are good with my latest commit.

**Comment by aparcar** [comment]
I like the new ordering, for me it makes sense due to the high amount of sub-packages.

**Comment by Tycale** [comment]
Thanks!

**Pull request merged**